Readme overhaul

pavlenex opened the pull request:
Closes #738 

Besides  of looking at diff, I think this readme can be [tested here on my fork a](https://github.com/pavlenex/stratum/tree/readme-update?tab=readme-ov-file#------stratum-v2-reference-implementation-sri--)s well, to get a better feeling on the visual changes.

I've removed a lot of stuff and tried to keep it consistent, if I removed something important, let me know. I believe a lot of what was removed was outdated or too specific and instead better suited for dev docs.

One thing I failed to figure out how to do properly is "Supporters" logos, I hope someone can help me with aligning them properly and scaling. 

Looking forward to feedback!

GitGab19 commented on README.md:
```diff
@@ -1,545 +1,132 @@
+  <a href="#donate">Donate</a> •
+  <a href="#supporters">Supporters</a> •
+  <a href="#license">License</a> 
+  <a href="msrv">MSRV</a>
```
I think here one '#' is missing (?)
The link is not working on your fork

pavlenex replied:
Good catch, I've fixed this

GitGab19 commented on README.md:
```diff
@@ -1,545 +1,132 @@
+</p>
 
-The goals of this project are to provide:
+## 👋Introduction
```
I would introduce a space between emoji and text

GitGab19 commented on README.md:
```diff
@@ -1,545 +1,132 @@
-flags.
-
-**Internal dependencies**:
+### 👷Miners
```
I would add a space in between also here

GitGab19 commented on README.md:
```diff
@@ -1,545 +1,132 @@
-- `binary_sv2` (maybe in the future will use the one already imported by `codec_sv2`)
-- `subprotocols/common_messages_sv2`
-- `subprotocols/template_distribution_sv2`
+- Sv1 Miners can use the translator proxy (`roles/translator`) to connect with a Sv2-compatible pool.
+- Sv1 mining farms mining to a Sv2-compatible pool gain some of the security and efficiency improvements Sv2 offers over Stratum V1 (Sv1). The Sv1<->Sv2 translator proxy does not support  _all_ the features of Sv2, but works as a temporary measure before upgrading completely to Sv2-compatible firmware. (The Sv1<->Sv2 translation proxy implementation is a work in progress.)
```
Uppercase for SV1 and SV2 (?)

pavlenex replied:
Good catch, done.

GitGab19 commented on README.md:
```diff
@@ -1,545 +1,132 @@
+- Sv1 mining farms mining to a Sv2-compatible pool gain some of the security and efficiency improvements Sv2 offers over Stratum V1 (Sv1). The Sv1<->Sv2 translator proxy does not support  _all_ the features of Sv2, but works as a temporary measure before upgrading completely to Sv2-compatible firmware. (The Sv1<->Sv2 translation proxy implementation is a work in progress.)
 
-### 3.10 protocols/v2/noise-sv2
+### 🛠️Pools
```
Space in between

GitGab19 commented on README.md:
```diff
@@ -1,545 +1,132 @@
 
-Contains the Noise Protocol logic.
-TODO: More information.
+- Pools supporting Sv2 can deploy the open source binary crate (`roles/pool`) to offer their clients (miners participating in said pool) an Sv2-compatible pool.
+- The Rust helper library provides a suite of tools for mining pools to build custom Sv2 compatible pool implementations.
+- The C library provides a set of FFI bindings to the Rust helper library for miners to integrate Sv2 into their existing firmware stack.
```
Uppercase SV2 (?)

GitGab19 commented on README.md:
```diff
@@ -1,545 +1,132 @@
 
-**External dependencies**: TODO
-**Internal dependencies**: TODO
+## 🛣Roadmap 
```
Space

GitGab19 commented on README.md:
```diff
@@ -1,545 +1,132 @@
-- job logic (this overlaps with the channel logic)
-- Bitcoin data structures <-> Sv2 data structures mapping
-- utils
+### 🎯Goals
```
Space

GitGab19 commented on README.md:
```diff
@@ -1,545 +1,132 @@
+The goals of this project are to provide:
 
-To build the various projects/examples follow the following commands:
+1. A robust set of Stratum V2 (Sv2) primitives as Rust library crates which anyone can use
+   to expand the protocol or implement a role. For example:
+   - Pools supporting Sv2
+   - Mining-device/hashrate producers integrating Sv2 into their firmware
+   - Bitcoin nodes implementing Template Provider to build the `blocktemplate`
+2. The above Rust primitives as a C library available for use in other languages via FFI.
+3. A set of helpers built on top of the above primitives and the external Bitcoin-related Rust crates for anyone to implement the Sv2 roles.
+4. An open-source implementation of a Sv2 proxy for miners.
+5. An open-source implementation of a Sv2 pool for mining pool operators.
```
Uppercase SV2 (?)

GitGab19 commented on README.md:
```diff
@@ -1,545 +1,132 @@
+5. An open-source implementation of a Sv2 pool for mining pool operators.
 
-`cargo build` - This quickly builds a test/debug artifacts for all crates
+## 💻Contribute 
```
Space

GitGab19 commented on README.md:
```diff
@@ -1,545 +1,132 @@
- | with_[tokio](https://tokio.rs/)                       | For async with custom (de)serializer                                                                | no      | network_helpers                                                                                              |
- | debug                                                 | Turns on debugging features                                                                         | no      | utils/buffer                                                                                                 |
- | [criterion](https://github.com/bheisler/criterion.rs) | To [run benchmark tests](utils/buffer/README.md)                                                    | no      | utils/buffer                                                                                                 |
+## 🤝Support
```
Space

GitGab19 commented on README.md:
```diff
@@ -1,545 +1,132 @@
+## 🤝Support
 
-#### Building/Updating sv2-ffi
+Join our Discord community to get help, share your ideas, or discuss anything related to Stratum V2 and its refference implementation. 
```
Typo: reference

GitGab19 commented on README.md:
```diff
@@ -1,545 +1,132 @@
+[Join the Stratum V2 Discord Community](https://discord.gg/fsEW23wFYs)
 
-### 4.02 Test
+## 🎁Donate
```
Space

GitGab19 commented on README.md:
```diff
@@ -1,545 +1,132 @@
+## 🎁Donate
 
-As a prerequisite, you must have [`cargo-tarpaulin`](https://github.com/xd009642/tarpaulin) installed globally:
+### 👤Individual Donations 
```
Space

GitGab19 commented on README.md:
````diff
@@ -1,545 +1,132 @@
-```
-$ cargo +nightly tarpaulin --verbose
-```
+### 🏢Corporate Donations
````
Space

GitGab19 commented on README.md:
```diff
@@ -1,545 +1,132 @@
+Email us at: stratumv2@gmail.com
 
-### 4.03 Run
+## 🙏Supporters
```
Space

GitGab19 commented on README.md:
```diff
@@ -1,545 +1,132 @@
 
-To get a list of the available binaries that can be run type `cargo run --bin` from the root. Then to run specify the 
-binary - for example from the table below to run the interop-cpp test you'd run `cargo run --bin interop-cpp`
+SRI contributors are independently, financually supported by following entities: 
```
Typo: financially

pavlenex replied:
Good catch, fixed

GitGab19 commented on README.md:
Space

GitGab19 reviewed (commented):
I really like the new structure, it's very clear for externals!
I left some typos and corrections/suggestion

pavlenex commented:
@GitGab19 Thanks for a review, I've added spacing with emojies, the only compromise there is that I had to add an ugly link for getting started guide, but that's a minor thing. I've also fixed few typos I found and the one you reported.

pavlenex commented:
@Fi3 when you get a chance, review and merge.
